test/test_prototype_transforms_consistency.py

```diff
@@ -840,3 +840,681 @@ def test_aa(self, inpt, interpolation):
+            target["masks"] = features.SegmentationMask(torch.randint(0, 2, size=(22, *size), dtype=torch.long))
+
+        yield (pil_image, target)
+
+        tensor_image = torch.randint(0, 256, size=(3, *size), dtype=torch.uint8)
+        target = {
+            "boxes": make_bounding_box(image_size=size, format="XYXY", extra_dims=(22,), dtype=torch.float),
+            "labels": features.Label(torch.randint(0, 81, size=(22,))),
+        }
+        if with_mask:
+            target["masks"] = features.SegmentationMask(torch.randint(0, 2, size=(22, *size), dtype=torch.long))
+
+        yield (tensor_image, target)
+
+        feature_image = features.Image(torch.randint(0, 256, size=(3, *size), dtype=torch.uint8))
+        target = {
+            "boxes": make_bounding_box(image_size=size, format="XYXY", extra_dims=(22,), dtype=torch.float),
+            "labels": features.Label(torch.randint(0, 81, size=(22,))),
+        }
+        if with_mask:
+            target["masks"] = features.SegmentationMask(torch.randint(0, 2, size=(22, *size), dtype=torch.long))
+
+        yield (feature_image, target)
+
+    def _test_transform(self, t_ref, t, data_kwargs={}):
+        for dp in self.make_datapoints(**data_kwargs):
+
+            # We should use prototype transform first as reference transform performs inplace target update
+            torch.manual_seed(12)
+            output = t(dp)
+
+            torch.manual_seed(12)
+            expected_output = t_ref(*dp)
+
+            assert_equal(expected_output, output)
+
+    def test_randomhorizontalflip(self):
+        t_ref = RandomHorizontalFlip(p=1.0)
+        t = prototype_transforms.RandomHorizontalFlip(p=1.0)
+        self._test_transform(t_ref, t)
+
+    def test_randomioucrop(self):
+        t_ref = RandomIoUCrop()
+        t = prototype_transforms.RandomIoUCrop()
+        self._test_transform(t_ref, t, {"with_mask": False})
+
+    def test_randomzoomout(self):
+        t_ref = RandomZoomOut()
+        t = prototype_transforms.RandomZoomOut()
+        self._test_transform(t_ref, t, {"with_mask": False})
+
+    def test_scalejitter(self):
+        t_ref = ScaleJitter((1024, 1024))
+        t = prototype_transforms.ScaleJitter((1024, 1024))
+        self._test_transform(t_ref, t)
+
+    def test_fixedsizecrop(self):
+        t_ref = FixedSizeCrop(size=(1024, 1024), fill=0)
+        t = prototype_transforms.FixedSizeCrop(size=(1024, 1024), fill=0)
+        self._test_transform(t_ref, t)
+
+    def test_randomshortestsize(self):
+        t_ref = RandomShortestSize(min_size=(480, 512, 544, 576, 608, 640, 672, 704, 736, 768, 800), max_size=1333)
+        t = prototype_transforms.RandomShortestSize(
+            min_size=(480, 512, 544, 576, 608, 640, 672, 704, 736, 768, 800), max_size=1333
+        )
+        self._test_transform(t_ref, t)
+
+
+# -----
+# Dumped reference detection transforms here for consistency checks
+# torchvision/references/detection/transforms.py
+# -----
```

Comment:
Instead of dumping, we should just import them. This avoid duplication and enforces the tests to be in sync with the actual references:

```py
import pathlib
import importlib.util


def import_reference_module(*parts):
    *path_parts, name = parts

    here = pathlib.Path(__file__).parent
    project_root = here.parent
    location = project_root.joinpath("references", *path_parts, f"{name}.py")

    spec = importlib.util.spec_from_file_location(name, location=location)
    module = importlib.util.module_from_spec(spec)
    spec.loader.exec_module(module)
    return module


detection_transforms = import_reference_module("detection", "transforms")
t_ref = detection_transforms.RandomHorizontalFlip(p=1.0)
```

Reply:
I dumped it there as it was the easiest way to do especially that we wont keep this test file in the future...

Reply:
> we wont keep this test file in the future...

True, but the two sources can still go out of sync until then. Plus, only ~100 lines in this PR is actually testing and the other ~700 lines are just copy pasted. This makes this file quite large and harder to work with for no reason.

Reply:
OK, i'll use importilb to get the code